main.go

```diff
@@ -409,3 +416,44 @@ func isCRDDeployed(mgr ctrlmgr.Manager, gvr schema.GroupVersionResource) (bool,
 func concurrency(c int) controller.Options {
 	return controller.Options{MaxConcurrentReconciles: c}
 }
+
+func setupRemoteClusterCacheTracker(ctx *context.ControllerManagerContext, mgr ctrlmgr.Manager) (*remote.ClusterCacheTracker, error) {
+	secretCachingClient, err := client.New(mgr.GetConfig(), client.Options{
+		HTTPClient: mgr.GetHTTPClient(),
+		Cache: &client.CacheOptions{
+			Reader: mgr.GetCache(),
+		},
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to create secret caching client")
+		return nil, err
+	}
+
+	// Set up a ClusterCacheTracker and ClusterCacheReconciler to provide to controllers
+	// requiring a connection to a remote cluster
+	log := ctrl.Log.WithName("remote").WithName("ClusterCacheTracker")
+	tracker, err := remote.NewClusterCacheTracker(
+		mgr,
+		remote.ClusterCacheTrackerOptions{
+			SecretCachingClient: secretCachingClient,
+			ControllerName:      controllerName,
+			Log:                 &log,
+			Indexes:             []remote.Index{remote.NodeProviderIDIndex},
+		},
+	)
+	if err != nil {
+		setupLog.Error(err, "unable to create cluster cache tracker")
+		return nil, err
+	}
+
+	if err := (&remote.ClusterCacheReconciler{
+		Client:           mgr.GetClient(),
+		Tracker:          tracker,
+		WatchFilterValue: managerOpts.WatchFilterValue,
+	}).SetupWithManager(ctx, mgr, concurrency(vSphereClusterConcurrency)); err != nil {
```

Comment:
Wondering if that's the right concurrency to use.

We would pass the tracker to:
* VMController (which uses `vSphereVMConcurrency`) 
* ServiceAccountProviderController (which uses `providerServiceAccountConcurrency`)
* ServiceDiscoveryController (which uses `serviceDiscoveryConcurrency`)

@sbueringer I think we need here a separate concurrency flag, ideally it should get set the same as for the core CAPI controllers cluster concurrency? Or is this also a gap in core CAPI (I think no, not there)?, Is there any guidance?

Reply:
The Core CAPI controller uses the flag for cluster concurrency as it's related to cache the cluster.
Do we need a separate flag in CAPV?

Reply:
I would use a separate flag

Would also be better if we have a separate flag in core CAPI to be honest

Let's make an example. You have a mgmt cluster with 2k+ workload clusters. The cluster controller is doing some heavy lifting so you would want to use more workers (for 2k 10 should be still fine but it's just an example :D).

So if you want to increase the concurrency of the cluster controller you don't want the side effect that the cluster controller of the cluster cache tracker also gets more workers.


(Basically every single controller should have it's own concurrency flag, in core CAPI we re-use a bunch of them, @chrischdi can you follow-up in core CAPI?)

Reply:
Used the same flag as CAPI PR.